COPYRIGHT

```diff
@@ -37,3 +37,18 @@ warranty express or implied, or assumes any legal liability or responsibility
 for the accuracy, completeness, or usefulness of any information, apparatus,
 product, or process disclosed, or represents that its use would not infringe
 privately owned rights.
+
+			   EXTERNAL CONTRIBUTIONS
+
+Portions of this code have been contributed under the above license by:
+
+ * Intel Corporation
+ * Cray
+ * IBM Corporation
+ * Microsoft Corporation
+ * Mellanox Technologies Ltd.
+ * DataDirect Networks.
+ * Oak Ridge National Laboratory
+ * Sun Microsystems, Lustre group
+ * Dolphin Interconnect Solutions Inc.
+ * Institut Polytechnique de Bordeaux
```

Comment:
```suggestion
 * Institut Polytechnique de Bordeaux
 
 Contributions copyright their original authors.
```

This is one thing that's missing if we're trying to replicate the per-file headers. I think @pavanbalaji has another version of this text that might be better.

Reply:
I'll add it in shortly.